crates/net/network/src/transactions.rs

```diff
@@ -492,7 +507,7 @@ where
     fn on_bad_import(&mut self, hash: TxHash) {
         if let Some(peers) = self.transactions_by_peers.remove(&hash) {
             for peer_id in peers {
-                self.report_bad_message(peer_id);
+                self.on_request_error(peer_id, RequestError::BadResponse);
```

Comment:
last nit: we want to call the reputation kind function here, with `BadTransactions`